targets/TARGET_Realtek/TARGET_AMEBA/trng_api.c

```diff
@@ -0,0 +1,68 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2013-2016 Realtek Semiconductor Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "trng_api.h"
+#include "analogin_api.h"
+#include "analogin_ext.h"
+
+#ifdef DEVICE_TRNG
+static analogin_t tradcng;
```

Comment:
Current TRNG relies on ADC to generate random values *shrug*

The calling sequence should be as follows, tradcng should be properly inited and freed.

trng_init
trng_get_bytes
trng_free

But, I think fold it in trng_s should be a better approach.